### PR TITLE
Qa test jest

### PR DESCRIPTION
**Pull request by gdvl777**
- Use jest framework for testing.
- Create a test file ([..].test.js) for a file containing the add item and delete item functions that you want to test.
- Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be straightforward. Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks:
- Mock a storage object to "imitate" localStorage operations (only if you are using jest < 24.x, in newer versions localStorage is mocked automatically).
- Mock HTML to test if add/delete functions add or remove exactly one <li> element to/from the list in the DOM.
- Make sure you group your tests using the describe() method.

Optional requirement

- Add a --coverage flag to your test script in package.json. That way you will see the percentage coverage of your application code every time you run a test.

### CODE REVIEW

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @, 

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] This is a peer-programming project milestone and This project's collaborators are supposed to be Daniel Villalba Lynch (@gdvl777) and James Abah, not Daniel Villalba Lynch (@gdvl777) and Esteban Palacios (@Estete9), So please submit the correct pull request with all project requirements and the correct team members.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1,291 +1,183 @@
...
-- LinkedIn: [LinkedIn](https://linkedin.com/in/linkedinhandle)
+-   GitHub: [@gdvl777](https://github.com/gdvl777)
+-   LinkedIn: [LinkedIn](https://www.linkedin.com/)
 
 👤 **Author2**
 
-- GitHub: [@githubhandle](https://github.com/githubhandle)
-- Twitter: [@twitterhandle](https://twitter.com/twitterhandle)
-- LinkedIn: [LinkedIn](https://linkedin.com/in/linkedinhandle)
+- GitHub: [@EngruugbeJimmy](github.com/EngruugbeJimmy)
```
Comment (Efiamotu-1) [inline README.md]: - [x] This link in your readme is currently broken as it throws up an error 404 page please kindly provide the appropriate link to this as it is unprofessional to leave broken links in your readme file 👍

File: src/___tests___/taskFunctions.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// eslint-disable-next-line no-unused-vars
```
Comment (Efiamotu-1) [inline src/___tests___/taskFunctions.test.js]: - [ ] Please kindly fix the eslint error you are getting rather than disabling it 👍 

[OPTIONAL] 
- [x] I want to also suggest you add more test cases to the ones you have, each for the add and remove functions, also you can add test cases that covers edge cases like when a user tries to add a task without inputing any text in the provided fields, which can be represented by using an empty string `" "`

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @gdvl777  & @EngruugbeJimmy , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @gdvl777 ,

Great job so far!  

May I highlight the following: 
- [x]  You have a descriptive and professional Pull request👏  ✔️ 
- [x] Your code is clean and well organized✔️
- [x] Well  done implementing all the required changes✔️



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)
